infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -411,24 +428,26 @@ private fun findNearestLocationTrack(
         }
 }
 
-private fun createFeatureGeometry(
-    responseSettings: FrameConverterResponseSettingsV1,
-    point: AlignmentPoint,
-): GeoJsonGeometryPoint {
-    return if (FrameConverterResponseSettingV1.FeatureGeometry in responseSettings) {
-        GeoJsonGeometryPoint(coordinates = listOf(point.x, point.y))
+private fun createFeatureGeometry(params: FrameConverterQueryParamsV1, point: AlignmentPoint): GeoJsonGeometryPoint {
+    return if (params.featureGeometry) {
+        pointToFrameConverterCoordinate(params, point).let { coordinate ->
+            GeoJsonGeometryPoint(coordinates = listOf(coordinate.x, coordinate.y))
+        }
     } else {
         GeoJsonGeometryPoint.empty()
     }
 }
 
 private fun createSimpleFeatureMatchOrNull(
-    responseSettings: FrameConverterResponseSettingsV1,
+    params: FrameConverterQueryParamsV1,
     point: AlignmentPoint,
     distanceToClosestPoint: Double,
-): FeatureMatchSimpleV1? {
-    return if (FrameConverterResponseSettingV1.FeatureMatchSimple in responseSettings) {
-        FeatureMatchSimpleV1(x = point.x, y = point.y, distanceFromRequestPoint = distanceToClosestPoint)
+): FeatureMatchBasicV1? {
+    return if (params.featureBasic) {
+        FeatureMatchBasicV1(
+            coordinate = pointToFrameConverterCoordinate(params, point),
+            distanceFromRequestPoint = distanceToClosestPoint,
+        )
```

Comment:
Nykyrakenne: Jokainen palautettava dataluokka sisältää välittömästi koordinaatit oikeassa järjestelmässä. Probleema: edellyttää koordinaattimuunnoksen tekemistä (ja palautusarvoasetusten passaamista) eri paikkoihin. Näitä ei toistaiseksi ole kuin pari, mutta sinänsä pieni riski. Koodarin pitää siis muistaa kutsua tuota pointToFrameConverterCoordinate jokaisen paluutietorakenteen luonnin yhteydessä, joka edellyttää mahdollisen koordinaattimuunnoksen tekemistä.

Vaihtoehtoinen rakenne: Luodaan palautettavat dataluokat aluksi layout-koordinaateilla, ja mäpätään jokainen koordinaatti esim. controlleritasolla juuri ennen palautusta oikeaan formaattiin. Ongelma: Paluutyypit on nykyfunkkareilta yleisempää GeoJsonia, joten Controlleri ei nykyrakenteella tiedä suoraan mitä kenttiä tyypit sisältävät -> ei voida yksinkertaisesti etsiä ja mäpätä jokaista koordinaattia.

Mielipiteitä, muita idiksiä?

Reply:
Kinkkinen ongelma kyllä. Ei tule itsellekään mieleen mitään tämän fiksumpaa ratkaisua

Reply:
Joo olisi minusta mielekkäintä että tuon unohtamisesta tulisi käännösvirhe, mutta vähän hankalasti aikaansaatavissa ilman lisämiettimistä

Reply:
En ihan tajunnut että miten tuo muunnos voisi tällä nykyrakenteella unohtua. Eikös nuo paluutyypit pakota tuon (FrameConverterCoordinate) ainakin siinä määrin että ei se vahingossa jää muuntamatta? Tietenkin geojsoniksi muunnoksen jälkeen all bets are off